core/api/src/app/payments/helpers.ts

```diff
@@ -146,28 +146,38 @@ export const getPriceRatioForLimits = wrapAsyncToRunInSpan({
 })
 
 export const addContactsAfterSend = async ({
-  senderAccount,
-  recipientAccount,
+  senderContactEnabled,
+  senderAccountId,
+  senderUsername,
+
+  recipientContactEnabled,
+  recipientAccountId,
+  recipientUsername,
 }: {
-  senderAccount: Account
-  recipientAccount: Account
+  senderContactEnabled: boolean
+  senderAccountId: AccountId
+  senderUsername: Username | undefined
+
+  recipientContactEnabled: boolean
+  recipientAccountId: AccountId
+  recipientUsername: Username | undefined
 }): Promise<true | ApplicationError> => {
-  if (!(senderAccount.contactEnabled && recipientAccount.contactEnabled)) {
+  if (!(senderContactEnabled && recipientContactEnabled)) {
```

Comment:
if we move enabled fields at this level then does not make sense to do this validation here. The purpose of passing whole objects is to delegate to this method the business logic. (probably helpers is not a good location for this method)

Reply:
Fixed back here. I'd originally changed it because the function was inside the lock and I wanted to isolate the properties that could change (require potential account refetch), but since it's outside the lock now this is no longer relevant

https://github.com/GaloyMoney/galoy/pull/4181/commits/a1ffe7c8a0e522a8134d0ca3255aeb77c5533ceb

Reply:
Also I think you're right about this being a distinct interface being clearer if it weren't in helpers. Moved it to `accounts` module here https://github.com/GaloyMoney/galoy/pull/4181/commits/ba94cbbb40be988f046386b1e2de4c24063c2811